bundles/org.openhab.binding.freeathomesystem/src/main/java/org/openhab/binding/freeathomesystem/internal/handler/FreeAtHomeDeviceHandler.java

```diff
@@ -0,0 +1,588 @@
+import org.openhab.binding.freeathomesystem.internal.type.FreeAtHomeChannelTypeProvider;
+import org.openhab.binding.freeathomesystem.internal.util.FreeAtHomeHttpCommunicationException;
+import org.openhab.binding.freeathomesystem.internal.util.UidUtils;
+import org.openhab.binding.freeathomesystem.internal.valuestateconverter.ValueStateConverter;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.library.types.StopMoveType;
+import org.openhab.core.library.types.StringType;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.Channel;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.ThingUID;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.thing.binding.ThingHandler;
+import org.openhab.core.thing.binding.builder.ChannelBuilder;
+import org.openhab.core.thing.binding.builder.ThingBuilder;
+import org.openhab.core.thing.type.ChannelKind;
+import org.openhab.core.thing.type.ChannelType;
+import org.openhab.core.thing.type.ChannelTypeBuilder;
+import org.openhab.core.thing.type.ChannelTypeUID;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.openhab.core.types.State;
+import org.openhab.core.types.StateDescriptionFragmentBuilder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link FreeAtHomeDeviceHandler} is responsible for handling the generic free@home device main communication
+ * and thing updates
+ *
+ * @author Andras Uhrin - Initial contribution
+ *
+ */
+public class FreeAtHomeDeviceHandler extends BaseThingHandler {
+
+    private static final String CHANNEL_URI = "channel-type:freeathomesystem:config";
+
+    private final Logger logger = LoggerFactory.getLogger(FreeAtHomeDeviceHandler.class);
+    private FreeAtHomeDeviceDescription device = new FreeAtHomeDeviceDescription();
+    private FreeAtHomeChannelTypeProvider channelTypeProvider;
+
+    private String deviceID = null;
+
+    private static URI configDescriptionUriChannel = null;
+
+    private Map<ChannelUID, FreeAtHomeDatapointGroup> mapChannelUID = new HashMap<ChannelUID, FreeAtHomeDatapointGroup>();
+    private Map<ChannelUID, String> mapChannelUIDVal = new HashMap<ChannelUID, String>();
+
+    public FreeAtHomeDeviceHandler(Thing thing, FreeAtHomeChannelTypeProvider channelTypeProvider,
+            FreeAtHomeChannelGroupTypeProvider channelGroupTypeProvider) {
+        super(thing);
+
+        this.channelTypeProvider = channelTypeProvider;
+    }
+
+    @Override
+    public void initialize() {
+        Map<String, String> properties = getThing().getProperties();
+
+        deviceID = properties.get("deviceId");
+        ;
+
+        logger.debug("Start creating device - device id: {}", deviceID);
+
+        updateChannels();
```

Comment:
`initialize()` must be fast and non-blocking, so no network operations allowed: `reloadChannelTypes` -> `getFreeatHomeDeviceDescription`. You can schedule a job, and set status UNKNOWN before leaving `initialize()`.